docs/python/SQLAlchemy/_index.md

```diff
@@ -0,0 +1,204 @@
+        for row in result:
+            print(row)
+
+if __name__ == '__main__':
+    main()
+```
+#### With DB Framework
+```
+#!/usr/bin/env python3
+
+from sqlalchemy import create_engine, event
+from google.cloud.sqlcommenter.sqlalchemy.executor import BeforeExecuteFactory
+
+DB_URL = '...'  # DB connection info
+
+def main():
+    engine = create_engine(DB_URL)
+
+    listener = BeforeExecuteFactory(with_db_framework=True)
+    event.listen(engine, 'before_cursor_execute', listener, retval=True)
+
+    result = engine.execute('SELECT * FROM polls_question')
+    for row in result:
+        print(row)
+
+if __name__ == '__main__':
+    main()
+```
+
+#### With DB Driver
+```
+#!/usr/bin/env python3
+
+from sqlalchemy import create_engine, event
+from google.cloud.sqlcommenter.sqlalchemy.executor import BeforeExecuteFactory
+
+DB_URL = '...'  # DB connection info
+
+def main():
+    engine = create_engine(DB_URL)
+
+    listener = BeforeExecuteFactory(with_db_driver=True)
+    event.listen(engine, 'before_cursor_execute', listener, retval=True)
+
+    result = engine.execute('SELECT * FROM polls_question')
+    for row in result:
+        print(row)
+
+if __name__ == '__main__':
+    main()
+```
+
+```shell
+python3 main.py
+(1, 'Wassup?', datetime.datetime(2019, 5, 30, 13, 51, 12, 910545, tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=-420, name=None)))
+(2, 'Wassup?', datetime.datetime(2019, 5, 30, 13, 57, 45, 905771, tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=-420, name=None)))
+(3, 'Wassup?', datetime.datetime(2019, 5, 30, 13, 57, 46, 908185, tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=-420, name=None)))
+(4, 'Wassup?', datetime.datetime(2019, 5, 30, 13, 57, 47, 557196, tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=-420, name=None)))
+(5, 'Wassup?', datetime.datetime(2019, 5, 30, 13, 57, 47, 853424, tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=-420, name=None)))
+```
+
+### Results
+
+Examining our Postgresql server logs
+
+#### With OpenCensus
+```
+2019-07-18 14:10:15.228 -03 [30071] LOG:  statement: SELECT * FROM polls_question
+/*traceparent='00-bf66750ad4c76f614c0a99d843758cbb-e6b27c3caf35de73-01'*/
+```
+
+#### With DB Framework
+```
+2019-07-18 14:11:19.576 -03 [30108] LOG:  statement: SELECT * FROM polls_question
+/*db_framework='sqlalchemy%3A1.3.5'*/
+```
+
+
+#### With DB Driver
+```
+2019-07-18 14:03:33.426 -03 [29858] LOG:  statement: SELECT * FROM polls_question
+/*db_driver='psycopg2'*/
+```
+
+### With flask
+When coupled with the web framework [flask](http://flask.pocoo.org), we still provide middleware to correlate
+your web applications with your SQL statements from psycopg2. Please see this end-to-end guide below:<br>
+[![](/docs/images/flask-logo.png)](/docs/python/flask/_index.md##with-psycopg2)
+
+### References
+
+Resource|URL
+---|---
+sqlcommenter-sqlalchemy on PyPi|https://pypi.org/project/google-cloud-sqlcommenter
+sqlcommenter-sqlalchemy on Github|https://github.com/google/sqlcommenter
```

Comment:
Maybe we should point to the opentelemetry-sqlcommenter path ?

Reply:
Updated